plugins/Login/Controller.php

```diff
@@ -387,30 +387,32 @@ protected function authenticateAndRedirect($login, $password, $urlToRedirect = f
      */
     public function resetPassword()
     {
-        $infoMessage = null;
-        $formErrors = null;
-
         $form = new FormResetPassword();
-        if ($form->validate()) {
-            $nonce = $form->getSubmitValue('form_nonce');
-            $errorMessage = Nonce::verifyNonceWithErrorMessage('Login.login', $nonce);
-            if ($errorMessage === "") {
-                $formErrors = $this->resetPasswordFirstStep($form);
-                if (empty($formErrors)) {
-                    $infoMessage = Piwik::translate('Login_ConfirmationLinkSent');
-                }
-            } else {
-                $formErrors = [$errorMessage];
-            }
-        } else {
-            // if invalid, display error
-            $formData = $form->getFormData();
-            $formErrors = $formData['errors'];
+
+        if (false === $form->validate()) {
+            return $this->renderResetPasswordView($form->getFormData()['errors']);
         }
 
+        $nonceError = Nonce::verifyNonceWithErrorMessage('Login.login', $form->getSubmitValue('form_nonce'));
+
+        if (false === empty($nonceError)) {
+            return $this->renderResetPasswordView([$nonceError]);
+        }
+
+        $firstStepFormErrors = $this->resetPasswordFirstStep($form);
+
+        if (false === empty($firstStepFromErrors)) {
+            return $this->renderResetPasswordView([$firstStepFormErrors]);
+        }
```

Comment:
I found it harder than necessary to understand the long condition meant something is not empty, perhaps consider the `!empty` notation? Otherwise the refactoring of the method is nice and makes is easier to read.

```suggestion
        if (!empty($nonceError)) {
            return $this->renderResetPasswordView([$nonceError]);
        }

        $firstStepFormErrors = $this->resetPasswordFirstStep($form);

        if (!empty($firstStepFromErrors)) {
            return $this->renderResetPasswordView([$firstStepFormErrors]);
        }
```